manifests/client.pp

```diff
@@ -17,25 +19,52 @@
 #     package_name   => 'clickhouse-client-custom',
 #     package_ensure => 'hold',
 #     user           => 'custom-user',
-#     conf_d_dir     => '/some/path',
+#     config_d_dir     => '/some/path',
 #   }
 #
 # @author InnoGames GmbH
 #
 class clickhouse::client(
-    String[1]        $package_name   = 'clickhouse-client',
-    String[1]        $package_ensure = 'installed',
-    Stdlib::Unixpath $conf_d_dir     = '/etc/clickhouse-client/conf.d',
+    String[1]                  $package_name   = 'clickhouse-client',
+    String[1]                  $package_ensure = 'installed',
+    Optional[Stdlib::Unixpath] $conf_d_dir     = undef,
+    Stdlib::Unixpath           $config_d_dir   = $conf_d_dir ? {
+        undef   => '/etc/clickhouse-client/config.d',
+        default => $conf_d_dir,
+    },
 ) inherits clickhouse {
 
     package { $package_name:
         ensure => $package_ensure,
     }
 
-    file { $conf_d_dir:
-        ensure  => 'directory',
-        owner   => $clickhouse::user,
-        group   => $clickhouse::group,
-        require => Package[$package_name],
+    # TODO: remove it in 3 releases
+    if ($conf_d_dir) {
+        notify { 'conf_d deprecation':
+            message  => @("END")
+                Parameter conf_d_dir is deprecated, use config_d_dir
+                Be aware that after remove it the ${conf_d_dir} will be removed as well, you shouldn't use it.
+                See https://clickhouse.yandex/docs/en/operations/configuration_files
+                | END
+            ,
+            loglevel => 'warning',
+        }
+
+        file { $conf_d_dir:
+            ensure  => 'directory',
+            require => Package[$package_name],
+        }
+
+    } else {
+        file { '/etc/clickhouse-client/conf.d':
+            ensure => 'absent',
```

Comment:
User should decide by himself about puppetizing them

Reply:
how can he decide when you manage the directory?

Reply:
Let's continue in one discussion, here https://github.com/innogames/puppet-clickhouse/pull/4/files/f5374567adbff932bd6c71dece78a8a0ce457ce2..d54b607be668f93f47aa0527212e663d1e35e7c4#diff-6fa83dd25c6a65d4afde2eb252c8b945

Reply:
So, sorry. I mixed up different things.

I explicitly not set here `recurse`, `purge` and `force`, since conf.d used to be the default directories, and a user may have there manually managed configs, that we aren't supposed to touch.

Reply:
ah now i get it. agreed